speechbrain/lobes/models/huggingface_wav2vec.py

```diff
@@ -39,162 +40,445 @@
+            File.endswith(".bin") or File.endswith(".ckpt")
+            for File in os.listdir(str(sink))
+        ):
+            is_local = True
+            local_path = str(sink)
+        else:
+            local_path = path
+    else:
+        local_path = path
+
+    if is_local:
+        # Test for HuggingFace model
+        if any(File.endswith(".bin") for File in os.listdir(local_path)):
+            is_sb = False
+            return is_sb, checkpoint_filename
+
+        # Test for SpeechBrain model and get the filename.
+        for File in os.listdir(local_path):
+            if File.endswith(".ckpt"):
+                checkpoint_filename = os.path.join(path, File)
+                is_sb = True
+                return is_sb, checkpoint_filename
+    else:
+        files = model_info(path).siblings  # get the list of files of the Hub
+
+        # Test if it's an HuggingFace model or a SB one
+        for File in files:
+            if File.rfilename.endswith(".ckpt"):
+                checkpoint_filename = File.rfilename
+                is_sb = True
+                return is_sb, checkpoint_filename
 
-    Source paper wav2vec2.0: https://arxiv.org/abs/2006.11477
-    Source paper Hubert: https://arxiv.org/abs/2106.07447
-    Transformer from HuggingFace needs to be installed:
-    https://huggingface.co/transformers/installation.html
+        for File in files:
+            if File.rfilename.endswith(".bin"):
+                checkpoint_filename = File.rfilename
+                is_sb = False
+                return is_sb, checkpoint_filename
 
-    The model can be used as a fixed feature extractor or can be finetuned. It
-    will download automatically the model from HuggingFace or use a local path.
+    err_msg = f"{path} does not contain a .bin or .ckpt checkpoint !"
+    raise FileNotFoundError(err_msg)
+
+
+def config_return_hidden_states(config):
+    """Sets `output_hidden_states = True` for a transformer config.
+    """
+    config.output_hidden_states = True  # We want the hidden states as well!
+    return config
+
+
+def model_set_spectral_augmentation(model, apply_spec_augment):
+    """Sets `model.config.apply_spec_augment` the flag to a specific value.
+    """
+    model.config.apply_spec_augment = apply_spec_augment
+    return model
+
+
+def modify_state_dict_wav2vec2(path):
+    """Function to be used in HuggingFaceModel._load_sb_pretrained_parameters.
+
+    A custom loading ensures SpeechBrain compatibility for Pretrain and model
+    de/serialization. Here, '.wav2vec2' is removed before loading.
+    """
+    modified_state_dict = {}
+    orig_state_dict = torch.load(path, map_location="cpu")
+
+    # We remove the .wav2vec2 in the state dict.
+    for key, params in orig_state_dict.items():
+        if "wav2vec2." in key:
+            save_key = key.replace("model.wav2vec2.", "")
+            modified_state_dict[save_key] = params
+
+    return modified_state_dict
+
+
+def default_forward(model, data):
```

Comment:
Do you think we should add an example to better illustrate/have an automatic test?

Reply:
It's not intended for use in that way. The concept works here more contextual - infiltrate with the forward outer wrapper with the right snippet to insert for modification of function behaviour. It's a guided insertion attack :yum:

Reply:
done.

Testing is in the TODO list of the PR - yaml files etc - the default here migt not be called for the specific nature of each HF transformers model.